src/walktree.js

```diff
@@ -1,49 +1,60 @@
-const fs = require('fs');
+var fs = require('fs');
+var path = require('path');
 
-const walkTree = function walkTree(opts) {
-    const path = opts.root;
-    const ignored = opts.ignore;
-    const create = opts.create;
-    const countFolders = opts.countFoldersAsFiles;
-    const fileName = opts.file;
 
-    walkTreeRecursive(path);
+function walkTree (opts) {
+    var create = opts.create;
+    var ignored = opts.ignore;
+    var countFolders = opts.countFoldersAsFiles;
+    var nameOfPlaceholder = opts.file;
 
-    function walkTreeRecursive(path) {
-        let elements = fs.readdirSync(path);
-        let fileCount = 0;
+    walkTreeRecursive(opts.root);
 
-        while (elements.length > 0) {
-            let element = elements.pop();
-            let elementPath = path + '/' + element;
-
-            // Skip ignored folder and tries next folder in line
-            if (ignored.includes(element))
-                continue;
+    function walkTreeRecursive(dirPath) {
+        fs.readdir(dirPath, function(err, elements) {
+            if (err) {
+              throw new Error('Failed to read the directory at: ' + dirPath);
+            }
```

Comment:
Error checking is important, especially when a user's folder might not be the cleanest thing in the world, and file system errors can pop up anywhere.